src/components/app-content-renderer/use-request.js

```diff
@@ -8,13 +8,17 @@ const useRequest = (data, onResponse, onError) => {
   const [state, setState] = useReducer(reducer, {
     loaded: typeof data.count === 'number',
     count: data.count,
+    ...data,
+    ...data?.shape,
   });
 
   useEffect(async () => {
     try {
-      const response = await processRequest(data);
-      setState({ loaded: true, ...response });
-      onResponse && onResponse(response, data);
+      if (typeof state.count === 'undefined') {
+        const response = await processRequest(data);
+        setState({ loaded: true, ...response });
+        onResponse && onResponse(response, data);
```

Comment:
Can we use safe accessor?
```suggestion
        onResponse?.(response, data);
```

Reply:
Sorry I pressed the button to quickly :face_exhaling: